include/zyre_library.h

```diff
@@ -63,6 +63,14 @@
 #   endif
 #endif
 
+#ifdef NDEBUG
```

Comment:
This shouldn't be in the public header - it will have an unpredictable effect on downstream applications, possibly breaking them. Could you move it to an internal header?

Reply:
Good point. Do you think its worth having this for each zproject based project in their private headers. After all our recommendation is not to disable asserts for release builds. Why not enforce it.

Reply:
Yes, that makes sense to me